src/mame/recfranco/rf51w14cg.cpp

```diff
@@ -109,15 +112,65 @@ static INPUT_PORTS_START( rf51w14cg )
 	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
 INPUT_PORTS_END
 
+// Only two dip switches banks, both on CPU PCB
+static INPUT_PORTS_START( rfsantafem )
+	PORT_START("DSW1")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+
+	PORT_START("DSW2")
+	PORT_DIPUNKNOWN_DIPLOC(0x01, 0x01, "SW1:1")
+	PORT_DIPUNKNOWN_DIPLOC(0x02, 0x02, "SW1:2")
+	PORT_DIPUNKNOWN_DIPLOC(0x04, 0x04, "SW1:3")
+	PORT_DIPUNKNOWN_DIPLOC(0x08, 0x08, "SW1:4")
+	PORT_DIPUNKNOWN_DIPLOC(0x10, 0x10, "SW1:5")
+	PORT_DIPUNKNOWN_DIPLOC(0x20, 0x20, "SW1:6")
+	PORT_DIPUNKNOWN_DIPLOC(0x40, 0x40, "SW1:7")
+	PORT_DIPUNKNOWN_DIPLOC(0x80, 0x80, "SW1:8")
+INPUT_PORTS_END
+
 void rf51w14cg_state::rf51w14cg(machine_config &config)
 {
 	I80188(config, m_maincpu, 32_MHz_XTAL / 2); // Intel N80C188XL-20, guess divisor
 
+	I8255A(config, "pia1"); // OKI M82C55A
+	I8255A(config, "pia2"); // OKI M82C55A
+	I8255A(config, "pia3"); // OKI M82C55A
+	I8255A(config, "pia4"); // OKI M82C55A
+	I8255A(config, "pia5"); // OKI M82C55A
+	I8255A(config, "pia6"); // OKI M82C55A
+
 	SPEAKER(config, "mono").front_center();
 
 	OKIM6376(config, m_okim6376, XTAL(9'830'400)/64).add_route(ALL_OUTPUTS, "mono", 1.0); // Frecuency divisor as per JMP7
 }
 
+void rf51w14cg_state::rfsantafem(machine_config &config)
+{
+	I80188(config, m_maincpu, 32_MHz_XTAL / 2); // N80C188XL20
+
+	// PIAs on CPU PCB
+	I8255A(config, "pia1");
+	I8255A(config, "pia2");
+	I8255A(config, "pia3");
```

Comment:
Do the schematics call them "PIA"? If so: OK. If not: 8255 is PPI.
Same for above where you added 6 of them, and one more below.

Reply:
Fixed, thanks!